packages/calypso-build/typescript/ts-package.json

```diff
@@ -0,0 +1,34 @@
+{
+	"$schema": "https://json.schemastore.org/tsconfig",
+
+	"compilerOptions": {
+		// Allow composed projects
+		"composite": true,
+		"disableReferencedProjectLoad": true,
+
+		// Generate declarations and sourcemaps
+		"declaration": true,
+		"declarationMap": true,
+		"sourceMap": true,
+
+		// Do not emit helpers, rely on importing tslib
+		"importHelpers": true,
+		"noEmitHelpers": true,
+		"skipLibCheck": true,
+
+		// Target modern JS
+		"lib": [ "ES2020", "DOM" ],
+		"module": "ES2020",
+		"target": "ES2020",
+		"esModuleInterop": true,
+		"jsx": "react",
+
+		// Strict checks
+		"strict": true,
+		"isolatedModules": true,
+
+		"moduleResolution": "node",
+		"types": [ "node" ],
+		"forceConsistentCasingInFileNames": true
```

Comment:
Including the `DOM` looks like we're targeting browsers, but then we include `node` types which is for Node.

Are we building node packages? make two configs that extend from this one to target node or the browser. I wouldn't recommend targeting both environments.

Reply:
Many packages are using `process.env.*`, which Typescript wont find types for unless I add `types: [ "node" ]`

> Are we building node packages? make two configs that extend from this one to target node or the browser.

I started with that pattern, but it got blurry very quickly: our "browser" packages use Node types, and we don't have any TS package that targets Node (yet).

Reply:
Actually "many packages" it just 5 of them. I'll drop the `types:["node"]` here (so this config is only for browser packages) and add a `types:["node"]` in each package that needs it.

Reply:
Those are probably webpack-bundled packages. There's a [`@types/webpack-env`](https://www.npmjs.com/package/@types/webpack-env) package for that we could use. I'm not sure whether it's better behaved with things like the `NodeJS.Timeout` issue you mentioned.